### PR TITLE
Print more information about PKCS#12 structures

### PR DESCRIPTION
**Pull request by tomato42**
the pkcs12 tool can print more information about the decoded PKCS#12 file, so make it do it.

### CODE REVIEW

**Comment by richsalz** [comment]
+1.  @mattcaswell post-1.1.0 or not?

**Comment by mattcaswell** [comment]
Its a feature. Post 1.1.0.

**Comment by richsalz** [comment]
@tomato42 can you rebase and sign our CLA?

**Comment by tomato42** [comment]
where's the CLA? I don't see reference to it in `CONTRIBUTING`

**Comment by richsalz** [comment]
https://www.openssl.org/policies/cla.html

**Comment by levitte** [comment]
@tomato42, is there any progress on the CLA request?

**Comment by richsalz** [comment]
RedHat will not sign a CCLA, but will let individuals sign an ICLA.  @tomato42, ping Mark Cox if you need internal pointers.

**Comment by richsalz** [comment]
nudge.  @tomato42 .., :)

**Comment by tomato42** [comment]
completely forgot about this PR, will take a look at the CLA tomorrow

**Comment by tomato42** [comment]
CLA sent
during rebasing I noticed that scrypt is not handled correctly so I need to work on this PR a little bit more

**Comment by levitte** [comment]
Ok then.  Ping when it's ready for review

**Comment by tomato42** [comment]
@levitte ready

**Comment by tomato42** [comment]
am I reading the failure of
```
if [ -n "$CHECKDOCS" ]; then if $make doc-nits; then
 echo -e '+\057\057 MAKE DOC-NITS OK'; else echo -e '+\057\057 MAKE DOC-NITS FAILED'; false; fi; fi
```
correctly, in that it wants documentation for the new public methods?

**Comment by richsalz** [comment]
You can run "make doc-nits" yourself and see what it's complaining about. :)

**Comment by tomato42** [comment]
```
$ make doc-nits
(cd .; /usr/bin/perl util/find-doc-nits -n -p ) >doc-nits
if [ -s doc-nits ] ; then cat doc-nits; rm doc-nits ; exit 1; fi
*** WARNING: line containing nothing but whitespace in paragraph at line 19 in file doc/man3/OSSL_STORE_INFO.pod
*** WARNING: line containing nothing but whitespace in paragraph at line 55 in file doc/man3/OSSL_STORE_LOADER.pod
*** WARNING: line containing nothing but whitespace in paragraph at line 12 in file doc/man3/OSSL_STORE_open.pod
Makefile:450: recipe for target 'doc-nits' failed
make: *** [doc-nits] Error 1
```

I didn't break it... 😅

**Comment by richsalz** [comment]
I know who did :)

**Comment by tomato42** [comment]
the fix is trivial, should I add it to this PR?

**Comment by richsalz** [comment]
Rebase; master isn't broken.

**Comment by tomato42** [comment]
done

**Comment by tomato42** [comment]
@richsalz  are you sure that `master` isn't broken? I'm getting the exact same error and I'm at 0a3452520f which is the current `master` in github...

**Comment by richsalz** [comment]
My perl is old ☹  Please add a fix or make a new PR.  Thank you!

**Comment by InfoHunter** [comment]
That doc-nits are addressed in #3860

**Comment by InfoHunter** [comment]
But I am wondering why other PRs didn't trigger this...

**Comment by tomato42** [comment]
that PR isn't simple either, so I'm not sure if it will get merge quickly - I'd like to see Travis and AppVeyor builds passing and that's a blocker for it

**Comment by InfoHunter** [comment]
I also have made a same fix some days ago, but it seems without the fix, other PRs would not fail due to the 'doc-nits' in STORE docs. See #3882

**Comment by InfoHunter** [comment]
> so I'm not sure if it will get merge quickly

Just noticed this PR has been there for one year...

**Comment by tomato42** [comment]
What? #3860 is 21 days old with last comment being from 11 days ago...

**Comment by InfoHunter** [comment]
> PRs would not fail due to the 'doc-nits' in STORE docs.

Ah, just for no misunderstandings, 'not fail' here means only in travis-ci, I can also generate those nits locally : )

**Comment by InfoHunter** [comment]
> What? #3860 is 21 days old with last comment being from 11 days ago...

I meant your PR (#1334, this one), that surprised me a bit : )

**Comment by tomato42** [comment]
that's because of https://github.com/openssl/openssl/pull/1334#issuecomment-317267947 and me not having the time to go through the CLA earlier

**Comment by InfoHunter** [comment]
Any idea on this, @kaduk ?

**Comment by tomato42** [comment]
@levitte, @richsalz PR ready, passing all the automated checks

**Comment by InfoHunter** [comment]
> Any idea on this, @kaduk ?

Seems because this PR changes `util/libcrypto.num` file...

**Comment by tomato42** [comment]
previous versions didn't change it, I changed it so that AppVeyor CI did pass

File: apps/pkcs12.c [hunk 1]
```diff
@@ -523,12 +523,19 @@ int pkcs12_main(int argc, char **argv)
         const ASN1_INTEGER *tmaciter;
         const X509_ALGOR *macalgid;
         const ASN1_OBJECT *macobj;
-        PKCS12_get0_mac(NULL, &macalgid, NULL, &tmaciter, p12);
+        const ASN1_OCTET_STRING *tmac;
+        const ASN1_OCTET_STRING *tsalt;
+        PKCS12_get0_mac(&tmac, &macalgid, &tsalt, &tmaciter, p12);
```
Comment (richsalz) [inline apps/pkcs12.c]: blank line after declarations.

File: apps/pkcs12.c [hunk 2]
```diff
@@ -811,13 +818,28 @@ static int alg_print(const X509_ALGOR *alg)
             BIO_printf(bio_err, ", Iteration %ld, PRF %s",
                        ASN1_INTEGER_get(kdf->iter), OBJ_nid2sn(prfnid));
             PBKDF2PARAM_free(kdf);
+        } else if (pbenid == NID_id_scrypt) {
+            SCRYPT_PARAMS *kdf = NULL;
+            if (aparamtype == V_ASN1_SEQUENCE)
```
Comment (richsalz) [inline apps/pkcs12.c]: blank line.

**Comment by InfoHunter** [comment]
> previous versions didn't change it and were still failing

Strange...

**Comment by kaduk** [comment]
> idea on this @kaduk

#3860 needs another OMC review, ideally @levitte since it is trying to tweak things in a feature he introduced.
So far as I can tell, I have attempted to address all the comments it has received already.

**Comment by tomato42** [comment]
@richsalz nits fixed

**Comment by richsalz** [comment]
merged to master, did not squash.  thank you!

**Comment by tomato42** [comment]
You're welcome and thank you :)

**Pull request closed without merging**